### PR TITLE
README.rst: added docker usage

### PR DESCRIPTION
**Pull request by nitrocode**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -142,6 +142,16 @@ Given that, you can run Cloud Custodian with::
...
+You can run it with Docker as well
+
+  # Download the image
+  $ docker pull cloudcustodian/c7n
+
+  # Run the policy
+  $ docker run -it \
+      -v $(pwd)/policy.yml:/src/policy.yml \
+      --env-file <(env | grep "^AWS") \
+      cloudcustodian/c7n run -s out /src/policies/policy.yml
```
Comment (kapilt) [inline README.rst]: probably want a volume -v for the output directory as well. we also may need to cover on examples for gcp/azure,  I  can add those after the fact. its mostly just the specific credential file / env var needed for those providers.

Comment (nitrocode) [reply README.rst]: Good idea. I added `-v` to it.

Comment (nitrocode) [reply README.rst]: The policy was associated to AWS so I figured making one specific to AWS was okay but yeah examples for GCP and Azure would be nice too

**Comment by kapilt** [comment]
Thanks for the pull request!  If you don't mind could you sign the CLA at the bottom of the README? Direct link here: https://docs.google.com/forms/d/e/1FAIpQLSfwtl1s6KmpLhCY6CjiY8nFZshDwf_wrmNYx1ahpsNFXXmHKw/viewform

**Comment by nitrocode** [comment]
Signed!

**Pull request merged**